### PR TITLE
Feat/errorutil local allocation validation

### PR DESCRIPTION
**Pull request by SuryanshGarg04**
## Summary

This PR extends `errorutil check` to validate newly introduced error-code allocations while supporting both supported MeshKit development workflows:

* New errors using `"replace_me"` for allocation through the existing CI workflow.
* Developers running `errorutil update` locally to allocate an integer error code before committing.

Previously, newly introduced integer error codes were rejected outright. This change allows legitimate locally allocated integers while continuing to reject invalid manual assignments.

## What changed

* Added allocation-aware validation to `errorutil check`.
* Added support for baseline and current allocation summaries through:

  * `--baseline-summary`
  * `--current-summary`
* Validates newly introduced integer codes against the allocation window represented by the baseline and current `next_error_code` values.
* Detects allocation counter regression.
* Detects newly introduced duplicate integer error codes while tolerating duplicate codes already present in the baseline.
* Identifies existing integer codes by their numeric value rather than error constant name, so renames and moves preserving an existing code remain valid.
* Preserves the existing strict behavior when allocation summaries are not supplied.
* Added CLI and validation tests covering normal and edge-case allocation scenarios.

## Validation behavior

The allocation-aware validation requires newly introduced integer codes to satisfy:

```text
baselineNext <= code < currentNext
```

This means:

* valid locally allocated codes pass;
* arbitrary/manual integers outside the allocation window fail;
* allocation counter regression fails;
* newly introduced duplicate integer codes fail;
* duplicate codes already present in the baseline do not cause unrelated changes to fail;
* existing integer codes remain valid when an error is renamed or moved.

`errorutil check` remains validation-only. It does not allocate or modify error codes.

## Dependency

Depends on [[meshery/meshkit#1080](https://github.com/meshery/meshkit/pull/1080)](https://github.com/meshery/meshkit/pull/1080), which introduced the original `errorutil check` validation workflow.

The allocation-aware validator introduced here is consumed by downstream integration work in:

* [[meshery/meshery#21009](https://github.com/meshery/meshery/pull/21009)](https://github.com/meshery/meshery/pull/21009) — Meshery CI validation
* [[meshery/meshkit#1107](https://github.com/meshery/meshkit/pull/1107)](https://github.com/meshery/meshkit/pull/1107) — MeshKit CI and pre-commit validation

## Testing

Verified with:

* `go test ./cmd/errorutil/...`
* `go test ./...`
* `make check`
* CLI tests for summary-file handling and argument validation
* allocation scenarios including single and multiple local allocations
* rename handling
* duplicate-code handling
* counter regression
* allocation/deletion edge cases
* missing, malformed, and invalid summary metadata
* `git diff --check`

## Known limitations

This validation establishes consistency with the recorded allocation state; it does not provide cryptographic proof that a particular integer was produced by `errorutil update`.

It also does not provide global synchronization across independent branches that allocate from the same baseline.

## Summary by CodeRabbit

- **New Features**
  - The `check` command now accepts baseline and current summary files for validating allocation-counter changes.
  - Summary files are checked for valid `next_code` values and supported formats.

- **Bug Fixes**
  - Improved detection of duplicate numeric codes, invalid allocation ranges, counter regressions, and over-advanced counters.
  - Numeric codes are normalized consistently, with clearer diagnostics for invalid changes.

- **Documentation**
  - Updated command guidance to explain merge-base analysis and valid allocation paths.

### CODE REVIEW

File: cmd/errorutil/internal/coder/commands.go [hunk 1]
```diff
@@ -242,9 +352,49 @@ func commandCheck() *cobra.Command {
...
+				var cSum analysisSummary
+				if err := json.Unmarshal(cSumBytes, &cSum); err != nil {
+					return fmt.Errorf("failed to parse current summary: %w", err)
+				}
+				if cSum.NextCode <= 0 {
+					return fmt.Errorf("%s: next_code missing or zero — expected an errorutil_analyze_summary.json", currentSummaryPath)
+				}
+				currentNext = cSum.NextCode
+			} else if baselineSummaryPath != "" || currentSummaryPath != "" {
+				return fmt.Errorf("both --baseline-summary and --current-summary must be provided if one is provided")
```
Comment (lekaf974) [inline cmd/errorutil/internal/coder/commands.go]: this should be done in PreRunE and exit before entering RunE

File: cmd/errorutil/internal/coder/commands.go [hunk 2]
```diff
@@ -242,9 +352,49 @@ func commandCheck() *cobra.Command {
...
+					return fmt.Errorf("failed to read current summary: %w", err)
+				}
+				var cSum analysisSummary
+				if err := json.Unmarshal(cSumBytes, &cSum); err != nil {
+					return fmt.Errorf("failed to parse current summary: %w", err)
+				}
+				if cSum.NextCode <= 0 {
+					return fmt.Errorf("%s: next_code missing or zero — expected an errorutil_analyze_summary.json", currentSummaryPath)
+				}
+				currentNext = cSum.NextCode
```
Comment (lekaf974) [inline cmd/errorutil/internal/coder/commands.go]: Could we have a method to avoid duplicating this part since the same steps are done for baseline and current ?

File: cmd/errorutil/internal/coder/commands.go (hunk 1 above)
Comment (SuryanshGarg04) [reply cmd/errorutil/internal/coder/commands.go]: Yes. I’ll move this validation into PreRunE so the flag dependency is checked before entering RunE.

File: cmd/errorutil/internal/coder/commands.go (hunk 2 above)
Comment (SuryanshGarg04) [reply cmd/errorutil/internal/coder/commands.go]: Yes. I’ll extract the common summary reading and `next_code` validation into a helper and use it for both the baseline and current summaries, while preserving the error context.

File: cmd/errorutil/internal/coder/commands.go [hunk 3]
```diff
@@ -191,39 +193,170 @@ Meshery components and this tool:
...
 	return nil
 }
 
+func readSummaryNextCode(path, label string) (int, error) {
+	bytes, err := os.ReadFile(path)
+	if err != nil {
+		return 0, fmt.Errorf("failed to read %s summary: %w", label, err)
+	}
+	var sum struct {
+		NextCode int `json:"next_code"`
```
[the remark this answers is not in the document]
Comment (SuryanshGarg04) [reply cmd/errorutil/internal/coder/commands.go:334]: `next_code` is part of the existing summary JSON contract and is used consistently by the producer, consumer, CI workflow, and fixtures. Changing it to nextCode would break that contract, so I’m leaving it unchanged.

**Review by lekaf974 — APPROVED** [review]
